src/contrib/cluster/Akka.Cluster.Metrics/Serialization/MetricsGossip.cs

```diff
@@ -40,15 +37,15 @@ public MetricsGossip(IImmutableSet<NodeMetrics> nodes)
         /// </summary>
         public MetricsGossip Remove(Actor.Address node)
         {
-            return new MetricsGossip(Nodes.Where(n => !n.Address.Equals(node)).ToImmutableHashSet());
+            return new MetricsGossip(Nodes.Where(n => !n.Address.Equals(node)).ToImmutableHashSet(NodeMetricsComparer.Instance));
```

Comment:
This is needed so that the immutable hash set will use the comparer to compare hash code and equality between the NodeMetrics.